lawnchair/src/app/lawnchair/ui/preferences/components/AnnouncementPreference.kt

```diff
@@ -42,37 +44,49 @@ import app.lawnchair.ui.preferences.components.layout.PreferenceTemplate
 import app.lawnchair.ui.preferences.data.liveinfo.liveInformationManager
 import app.lawnchair.ui.preferences.data.liveinfo.model.Announcement
 import app.lawnchair.ui.util.addIf
-import com.android.launcher3.BuildConfig
 import com.android.launcher3.R
 import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
 
 @Composable
 fun AnnouncementPreference() {
     val liveInformationManager = liveInformationManager()
 
+    val coroutineScope = rememberCoroutineScope()
     val enabled by liveInformationManager.enabled.asState()
     val showAnnouncements by liveInformationManager.showAnnouncements.asState()
+    val dismissedAnnouncementIds by liveInformationManager.dismissedAnnouncementIds.asState()
     val liveInformation by liveInformationManager.liveInformation.asState()
 
     if (enabled && showAnnouncements) {
         AnnouncementPreference(
-            announcements = liveInformation.announcementsImmutable,
+            announcements = liveInformation.announcements.filter { it.id !in dismissedAnnouncementIds }.toImmutableList(),
```

Comment:
Please rebase your changes, and these calculations should be placed in `remember`.

Reply:
Done and done ✅